### PR TITLE
Unschedule pidgin tests from SLE

### PR DESCRIPTION
**Pull request by tjyrinki**
Unschedule pidgin tests from SLE, not included outside of PackageHub other than SLE <=12, and not included in current test runs there either.

- Related ticket: https://progress.opensuse.org/issues/94177
- Verification run: https://openqa.suse.de/tests/6288743

### CODE REVIEW

**Review by michaelgrifalconi — APPROVED** [review]
Thanks!

**Review by okurz — DISMISSED** [review]
What you put into the description of the pull request could also be in the commit message details. I don't know how you created your pull request but your git commit messages has *only* the subject line while the pull request description has more details. Please keep in mind that the github pull request description is only visible on github, the commit can be considered permanent information storage.

I can recommend the tool [hub](https://github.com/github/hub) (zypper in rubygem-hub) for easier PR creation. Also, I myself use a script [git-pr-last](http://github.com/okurz/scripts/tree/master/git-pr-last) to create a PR with proper description for these simple one-commit PRs:
```bash
$ cat $(which git-pr-last )
#!/bin/sh -e
target="${target:-"$USER"}"
git push $target && git show --no-patch --format=%B | hub pull-request -F -
```

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

**Comment by okurz** [comment]
Please get the approval of SUSE desktop QA in this PR as well, please

**Comment by michaelgrifalconi** [comment]
@okurz Probably not the right place to ask, but if you ask the approval of "SUSE Desktop QA" to remove this test, does this mean they own such test and I should have opened a bug for such team to either fix or remove the test?

**Comment by okurz** [comment]
> @okurz Probably not the right place to ask, but if you ask the approval of "SUSE Desktop QA" to remove this test, does this mean they own such test and I should have opened a bug for such team to either fix or remove the test? 

Well, the "ownership" in this case is not clearly defined however the test certainly also touches their domain of responsibility. And the test module maintainer as referenced in the source file defines who can be a good assignee for a ticket to remove/fix the test

**Comment by michaelgrifalconi** [comment]
In this case I would strongly suggest to first remove it and then talk about its possible re enablement. 

 This is breaking auto approval and there is a huge queue of stuck updates so I would remove all tests which are broken and require manual action from the reviewer.

**Comment by vpelcak** [comment]
@okurz It is not about the testcase being deleted. It is about our decision to not to use it for now or anymore.
While SUSE desktop QA would be affected by deletion, it certainly is not by our decision to unschedule.

Please also note, that as per agreements in our meetings in regard of openQA review change, squads are now expected to act fast on flaky testcases and free to unschedule the testcases.

Added value of tested functionality of pidgin IRC for operating system for servers is precisely zero.

Please proceed with unschedule.

**Comment by foursixnine** [comment]
> Please also note, that as per agreements in our meetings in regard of openQA review change, squads are now expected to act fast on flaky testcases and free to unschedule the testcases.

That agreement is not "achieved" or as far as I know, it's not official, but rather a proposal, If this is not correct, point me to the official email about it.

@michaelgrifalconi perhaps @GraceWang571 can shed some light wrt to the handling.

**Comment by tjyrinki** [comment]
Just noting I did already mention this to desktop QA asking them to schedule it separately if wanted, with us wishing to clean up it from main.pm in any case. They do not handle maintenance desktop QA at the moment unfortunately.

I won't be able to look at this further until next Tuesday earliest, so I'd recommend merging it to prevent continuation of breakage of test runs, but won't force it either.

**Comment by okurz** [comment]
> @okurz It is not about the testcase being deleted. It is about our decision to not to use it in our testing for now or anymore.
> While SUSE desktop QA would be affected by deletion, it certainly is not by our decision to unschedule.

How are they not affected if the PR removes the test from SLE test schedule in general? That's how I read the source change in lib/main_common.pm . The situation would be different if someone would remove the test module from a single YAML schedule file that by name tells that the test module would only be used within a restricted, specific scenario, e.g. if it would be "sle12sp3-functional.yml" or something.

> Please also note, that as per agreements in our meetings in regard of openQA review change, squads are now expected to act fast on flaky testcases and free to unschedule the testcases.

I agree with that but the according ticket https://progress.opensuse.org/issues/94177 is already set to resolved and I don't see an open ticket that says "ensure that the test module works as expected and add it back to tests". Often enough in the past I have seen tests being disabled, tickets resolved and nobody thought about eventually bringing back the test. As far as I know the process there should be a ticket for fixing the unscheduled test. Right?

> Added value of tested functionality of pidgin IRC for operating system for servers is precisely zero.

Well, I agree that there is no place for "pidgin IRC" on servers. However I am sure there is value for the test. There is a reason why the test was added in the first place and we have an enterprise desktop product that nowadays is equivalent to "server+WE module"

**Comment by okurz** [comment]
> Just noting I did already mention this to desktop QA asking them to schedule it separately if wanted, with us wishing to clean up it from main.pm SLE in any case. They do not handle maintenance desktop QA at the moment unfortunately.

@tjyrinki Please make such decisions *traceable*. From your commit it's not easily possible to find such documented decision. This is why I suggested to include your PR details in the git commit message, e.g. the progress ticket. But even https://progress.opensuse.org/issues/94177 does not mention that anywhere.

**Comment by michaelgrifalconi** [comment]
> @tjyrinki Please make such decisions _traceable_. From your commit it's not easily possible to find such documented decision. This is why I suggested to include your PR details in the git commit message, e.g. the progress ticket. But even https://progress.opensuse.org/issues/94177 does not mention that anywhere.

Commits merged via PR are traceable back to the PR. Example here https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/c6d49182c27a48bc23e5989e07c0dccb200ee0dc You can click on the link to the original PR.

**Comment by okurz** [comment]
> > @tjyrinki Please make such decisions _traceable_. From your commit it's not easily possible to find such documented decision. This is why I suggested to include your PR details in the git commit message, e.g. the progress ticket. But even https://progress.opensuse.org/issues/94177 does not mention that anywhere.
> 
> Commits merged via PR are traceable back to the PR. Example here [c6d4918](https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/c6d49182c27a48bc23e5989e07c0dccb200ee0dc) You can click on the link to the original PR right under the commit message.

Yes, I know how to use github :) However with this you are IMHO too much relying on github, make it harder to migrate and prevent the lookup in the git repo alone.

To be strict, effectively the git commit does not follow the https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style given to the project with approval of test maintainers. We need to stick to our own rules.

**Comment by michaelgrifalconi** [comment]
Good point. Now the commit message include poo and info. 

To be strict on openQA review side as well, after approving a lot of test manually due to this issue, I made sure QE-Core and Desktop QE are aware of the issue and I expect the two squads to solve the problem  (fix test or unschedule it) and I will not approve any update manually due to this failure.

My recommendation is still to unschedule (or fix it) before the weekend, to have a chance of some auto approvals since the update queue is still quite "clogged" at the moment.

**Review by okurz — APPROVED** [review]

**Comment by vpelcak** [comment]
> > @okurz It is not about the testcase being deleted. It is about our decision to not to use it in our testing for now or anymore.
> > While SUSE desktop QA would be affected by deletion, it certainly is not by our decision to unschedule.
> 
> How are they not affected if the PR removes the test from SLE test schedule in general? That's how I read the source change in lib/main_common.pm . The situation would be different if someone would remove the test module from a single YAML schedule file that by name tells that the test module would only be used within a restricted, specific scenario, e.g. if it would be "sle12sp3-functional.yml" or something.

I do not intend to interfere with new product validation. As long as only Maintenance: Test job groups have the testcase disabled, I am fine.

> > Please also note, that as per agreements in our meetings in regard of openQA review change, squads are now expected to act fast on flaky testcases and free to unschedule the testcases.
> 
> I agree with that but the according ticket https://progress.opensuse.org/issues/94177 is already set to resolved and I don't see an open ticket that says "ensure that the test module works as expected and add it back to tests". Often enough in the past I have seen tests being disabled, tickets resolved and nobody thought about eventually bringing back the test. As far as I know the process there should be a ticket for fixing the unscheduled test. Right?

The intention of the change is, that there is a tool which pokes the squads directly. Will it open poos, too?
If not, then openQA reviewers are not supposed to either.

> > Added value of tested functionality of pidgin IRC for operating system for servers is precisely zero.
> 
> Well, I agree that there is no place for "pidgin IRC" on servers. However I am sure there is value for the test. There is a reason why the test was added in the first place and we have an enterprise desktop product that nowadays is equivalent to "server+WE module"

I have strong reservations about practical usability of the test to our customers compared to work required to its maintenance. But that is not our call. That is on Timo and his squad.

**Comment by okurz** [comment]
> > I agree with that but the according ticket https://progress.opensuse.org/issues/94177 is already set to resolved and I don't see an open ticket that says "ensure that the test module works as expected and add it back to tests". Often enough in the past I have seen tests being disabled, tickets resolved and nobody thought about eventually bringing back the test. As far as I know the process there should be a ticket for fixing the unscheduled test. Right?
> 
> The intention of the change is, that there is a tool which pokes the squads directly. Will it open poos, too?
> If not, then openQA reviewers are not supposed to either.

My comment was not about any potential future process or tool with changed behaviour but practices and processes currently in place. As a reviewer within the project os-autoinst-distri-opensuse I want to verify that removing a test from a schedule is a conscious decision with good reasoning to prevent unexpected loss of coverage with the potential for regression leakage to users as test reviewers might still assume that a certain functionality of our products would be tested when in practice it's not anymore.

**Comment by tjyrinki** [comment]
We've a ticket now to re-schedule it (after being corrected it has not disappeared from SLE 15 WE either, my scripts had an omission), but it'll need some rework, probably switch to Libera Chat and reneedle etc. Anyway it's good to switch things from main pm towards YAML schedules for more granularity.

**Comment by okurz** [comment]
regarding switch to Libera Chat be aware of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12729

**Pull request merged**
